library/core/src/ptr/mod.rs

```diff
@@ -1451,9 +1451,11 @@ fnptr_impls_args! { A, B, C, D, E, F, G, H, I, J, K, L }
 /// as all other references. This macro can create a raw pointer *without* creating
 /// a reference first.
 ///
-/// Note, however, that the `expr` in `addr_of!(expr)` is still subject to all
-/// the usual rules. In particular, `addr_of!(*ptr::null())` is Undefined
-/// Behavior because it dereferences a null pointer.
+/// Note, however, that the `expr` in `addr_of!(expr)` is still subject to other rules:
```

Comment:
I think it would be a good idea to make it clear that these are just the normal rules for all `expr` in place context, and not something specific to `addr_of!`.

For example, `addr_of!(unsafe { unreachable_unchecked() })` is also still UB, of course. We don't want to list *all* language UB again here.